Examples/Swift/ManageOfflineRegionsExample.swift

```diff
@@ -0,0 +1,220 @@
+            if let error = error {
+                // Handle the error here if packs can't be removed.
+                print(error)
+            } else {
+                MGLOfflineStorage.shared.reloadPacks()
+            }
+        }
+    }
+
+    @objc func startOfflinePackDownload() {
+
+        /**
+         Create a region that includes the current map camera, to be captured
+         in an offline map. Note: Because tile count grows exponentially as zoom level
+         increases, you should be conservative with your `toZoomLevel` setting.
+         */
+        let region = MGLTilePyramidOfflineRegion(styleURL: mapView.styleURL,
+                                                 bounds: mapView.visibleCoordinateBounds,
+                                                 fromZoomLevel: mapView.zoomLevel,
+                                                 toZoomLevel: mapView.zoomLevel + 2)
+
+        // Store some data for identification purposes alongside the offline pack.
+        let userInfo = ["name": "\(region.bounds)"]
+        let context = NSKeyedArchiver.archivedData(withRootObject: userInfo)
+
+        // Create and register an offline pack with the shared offline storage object.
+        MGLOfflineStorage.shared.addPack(for: region, withContext: context) { (pack, error) in
+            guard error == nil else {
+                // Handle the error if the offline pack couldn’t be created.
+                print("Error: \(error?.localizedDescription ?? "unknown error")")
+                return
+            }
+
+            // Begin downloading the map for offline use.
+            pack!.resume()
+
+        }
+    }
+
+    // MARK: - MGLOfflinePack notification handlers
+
+    @objc func offlinePackProgressDidChange(notification: NSNotification) {
+        /**
+         Get the offline pack this notification is referring to,
+         along with its associated metadata.
+         */
+        if let pack = notification.object as? MGLOfflinePack,
+           let userInfo = NSKeyedUnarchiver.unarchiveObject(with: pack.context) as? [String: String] {
+
+            // At this point, the offline pack has finished downloading.
+            if pack.progress.countOfResourcesCompleted == pack.progress.countOfResourcesExpected {
+
+                let byteCount = ByteCountFormatter.string(fromByteCount: Int64(pack.progress.countOfBytesCompleted), countStyle: ByteCountFormatter.CountStyle.memory)
+
+                let packName = userInfo["name"] ?? "unknown"
+
+                print("""
+                  Offline pack “\(packName)” completed download:
+                    - Bytes: \(byteCount)
+                    - Resource count: \(pack.progress.countOfResourcesCompleted)")
+                """)
+
+            }
+        }
+        // Reload the table to update the progress percentage for each offline pack.
+        self.tableView.reloadData()
+
+    }
+
+}
+
+fileprivate extension MGLOfflinePackProgress {
```

Comment:
Lets move this below the table view methods.

Reply:
(Meaning, move `fileprivate extension MGLOfflinePackProgress {...}` below all the table view methods)